### PR TITLE
[Blaze i3] Intro screen

### LINKED ISSUE #10503: New Blaze intro screen

### PR DESCRIPTION
**Pull request by hichamboushaba**
Closes: #10503 

⚠️ Please don't merge until #10510 is merged

### Description
This PR adds the UI and logic of the campaign creation Intro screen, it's based on the work done on #10510 

### Testing instructions
For all the test cases please use a store with no Blaze campaigns

##### TC1: Store with a single published product
1. Open the app.
2. Trigger the blaze campaign creation (from MyStore or More menu)
3. Confirm the intro is shown.
4. Click on "Start your campaign"
5. Confirm a toast is shown.

##### TC2: Store with multiple published products
1. Open the app.
2. Trigger the blaze campaign creation (from MyStore or More menu)
3. Confirm the intro is shown.
4. Click on "Start your campaign"
5. Confirm the product selector is shown.
6. Select a product.
7. Confirm a toast is shown.

##### TC3: From product details
1. Open product details of a given product.
2. Tap on "promote with Blaze"
3. Confirm the intro is shown.
4. Click on "Start your campaign"
5. Confirm a toast is shown.

### Images/gif

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/intro/BlazeCampaignCreationIntroScreen.kt [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+            horizontalAlignment = Alignment.CenterHorizontally,
+            modifier = Modifier
+                .fillMaxWidth()
+                .weight(1f)
+                .verticalScroll(rememberScrollState())
+                .padding(dimensionResource(id = R.dimen.major_100))
+        ) {
+            Row(horizontalArrangement = Arrangement.spacedBy(dimensionResource(id = R.dimen.minor_100))) {
+                Icon(
+                    painter = painterResource(id = R.drawable.ic_blaze_monochrome),
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/intro/BlazeCampaignCreationIntroScreen.kt:123]: I tried tinting the existing blaze icon, but it didn't look right, so I added the monochrome version

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/intro/BlazeCampaignCreationIntroScreen.kt [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+                style = MaterialTheme.typography.h4,
+            )
+
+            Text(
+                text = stringResource(id = R.string.blaze_campaign_creation_new_intro_description),
+                style = MaterialTheme.typography.body1
+            )
+
+            Image(
+                painter = painterResource(id = R.drawable.img_blaze_intro),
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/intro/BlazeCampaignCreationIntroScreen.kt:149]: To confirm with Ann if the illustration looks right in the dark theme.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/intro/BlazeCampaignCreationIntroViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        launch {
+            if (navArgs.productId != -1L) {
+                triggerEvent(ShowCampaignCreationForm(navArgs.productId))
+            } else {
+                val products = getPublishedProducts()
+                when {
+                    products.size == 1 -> triggerEvent(ShowCampaignCreationForm(products.first().remoteId))
+                    products.isNotEmpty() -> triggerEvent(ShowProductSelector)
+                    else -> {
+                        WooLog.w(WooLog.T.BLAZE, "No products available to create a campaign")
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/intro/BlazeCampaignCreationIntroViewModel.kt:43]: This state is already managed in the previous PR, so this should not occur, but it's just here to be safe.

**Review by 0nko — COMMENTED** [review]
The My Store card shows a product to promote, however, when I start creating a campaign, the product list is empty (there are simple products in the store) 🤔 

[Screen_recording_20240103_155801.webm](https://github.com/woocommerce/woocommerce-android/assets/1522856/70f6b2a2-379a-4074-a5b4-4a0ee8509575)

**Review by hafizrahman — COMMENTED** [review]
The three test cases work correctly for me 💯

**Comment by hafizrahman** [comment]
Code also looks good to me, but I'm not sure what's wrong with Ondrej's case above. The product selector behavior is technically not a part of this PR 🤔

**Comment by JorgeMucientes** [comment]
Judging from the shared screenshot in the description there's a small UI glitch on the title and subtitle. They are both center-aligned in Figma, whereas they are not in the app: 

 



About the issue @0nko https://github.com/woocommerce/woocommerce-android/pull/10512#pullrequestreview-1802348830, I've experienced something similar while testing a different PR. When tapping on "Create campaign" nothing is opened/triggered. See the screen recording below: 

https://github.com/woocommerce/woocommerce-android/assets/2663464/e5a41daf-fcd4-4d6f-b879-ee3401aa436d

The bug is easy to reproduce: 
- Clear the app's data
- Log into a site that has products published and existing Blaze campaigns
- Do no open products tab
- Try to create a campaign from Blaze My Store view. If you had campaigns previously created, tapping on "Create campaign" won't do anything. 

The reason this happens is that in the scenario above, the product DB is empty, and we are not fetching the products from the API in `getPublishedProducts` from `BlazeCampaignCreationDispatcher`. This bug is unrelated to the changes from the PR. ~~I'll handle the fix in a different PR.~~ Fixed it here #10517

However, in @0nko's bug, the Product selector does fetch the products from the API after retrieving them from the DB, so I'm not sure why, in that case, the product selector shows empty results.

**Comment by JorgeMucientes** [comment]
Looks great! Good work everyone, :shipit:

**Review by JorgeMucientes — APPROVED** [review]

**Pull request merged**